gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -198,10 +198,18 @@ func (vm *VMKeeper) AddPackage(ctx sdk.Context, msg MsgAddPackage) error {
 			Context:   msgCtx,
 			MaxCycles: vm.maxCycles,
 		})
-	defer m2.Release()
-	m2.RunMemPackage(memPkg, true)
+	defer func() {
+		if r := recover(); r != nil {
+			err = errors.Wrap(fmt.Errorf("%v", r), "VM addpkg panic: %v\n%s\n",
+				r, m2.String())
+			consumeGas(ctx, m2, logPrefixAddPkg, pkgPath, "")
+			return
+		}
+		m2.Release()
+	}()
 
-	ctx.Logger().Info("CPUCYCLES", "addpkg", m2.Cycles)
+	m2.RunMemPackage(memPkg, true)
+	consumeGas(ctx, m2, logPrefixAddPkg, pkgPath, "")
```

Comment:
If the tx runs out of gas here, it will panic, so the `recover()` call below will not be nil, and so `consumerGas()` will be called a second time. Won't it panic again for the same reason?

If so you should make sure that the panic is not due to gas depletion before calling `consumerGas` in the `defer`. WDYT?